<?xml version="1.0"?>
<launch>
    <!-- <arg name="camera_topic_name" default="/photoneo_center/sensor/camera_info"/> -->
    <arg name="camera_topic_name" default="/photoneo_center/camera_info"/>
    
    <!-- <arg name="image_topic_name" default="/photoneo_center/sensor/image_color"/> -->
    <arg name="image_topic_name" default="/photoneo_center/rgb_texture"/>
    
    <!-- <arg name="input_cloud_topic_name" default="/cloud_without_segmented"/> -->
    <arg name="input_cloud_topic_name" default="/photoneo_center/pointcloud"/>
    
    <arg name="output_cloud_topic_name" default="/output"/>
    <arg name="target_frame" default=""/>
    <arg name="source_frame" default=""/>
    <arg name="paramter_output_file_path" default="/home/ericlab/dummy_cloud/paramter.txt"/>
    <!-- <arg name="f_scale" default="0.5"/> -->
    <!-- <arg name="cx_scale" default="0.99"/>
    <arg name="cy_scale" default="0.9"/> -->
    <arg name="f_scale" default="0.968"/>
    <arg name="cx_scale" default="1.01"/>
    <arg name="cy_scale" default="0.9"/>
    
    
    
    
    
    <node pkg="w2d_to_3d_ros" type="colored_pointcloud" name="colored_pointcloud" output="screen">
        <param name="camera_topic_name" value="$(arg camera_topic_name)"/>
        <param name="image_topic_name" value="$(arg image_topic_name)"/>
        <param name="input_cloud_topic_name" value="$(arg input_cloud_topic_name)"/>
        <param name="output_cloud_topic_name" value="$(arg output_cloud_topic_name)"/>
        <param name="target_frame" value="$(arg target_frame)"/>
        <param name="source_frame" value="$(arg source_frame)"/>
        <param name="paramter_output_file_path" value="$(arg paramter_output_file_path)"/>
        <param name="f_scale" value="$(arg f_scale)"/>
        <param name="cx_scale" value="$(arg cx_scale)"/>
        <param name="cy_scale" value="$(arg cy_scale)"/>
    </node>
</launch>